alvr/server_openvr/cpp/platform/linux/EncodePipeline.cpp

```diff
@@ -25,13 +25,14 @@ std::unique_ptr<alvr::EncodePipeline> alvr::EncodePipeline::Create(
     Renderer* render,
     VkContext& vk_ctx,
     VkFrame& input_frame,
-    VkFrameCtx& vk_frame_ctx,
+    VkImageCreateInfo& image_create_info,
     uint32_t width,
     uint32_t height
 ) {
     if (Settings::Instance().m_force_sw_encoding == false) {
         if (vk_ctx.nvidia) {
             try {
+                alvr::VkFrameCtx vk_frame_ctx(vk_ctx, image_create_info);
```

Comment:
small question, will this destruct itself when returning from this context?

Reply:
Yes, you'll need to store it as an member of `EncodePipeline`, ideally as an `std::optional`.

Reply:
it is a member EncodePipeline, stored later, here https://github.com/alvr-org/ALVR/pull/2664/files#diff-c05d15754786b3e214bb46c1c81d656d65987782d416f23c323ac8af7b7df2d6L35-L37 and here https://github.com/alvr-org/ALVR/pull/2664/files#diff-016ef3267bc3c690b686cfa31a35ba7a6fdabaad41d3489a346df0ac1d7ba9cbR45

Reply:
will put under std::optional

Reply:
Lemme take a look rq

Reply:
I was kinda stupid, it's way better to also modify the nvenc pipeline to take it as a normal, i.e. non reference parameter and just store it internally unconditionally.

Reply:
looking at code, std::optional looks unnecessary... wdym by storing as non-reference parameter? is current code ok?

Reply:
No

Reply:
I'll just directly push to the branch later

Reply:
i pushed some changes, i think you meant something like this